### ISSUE TITLE
Questions about your fine-tuning process.

### ISSUE THREAD

**Issue by MooreAndMoore** [Albert-Ma/PROP#7]
Hi, I recently read your new paper B-PROP in SIGIR'21. I'm trying to take your methods as baselines. I have some questions about your fine-tuning process as follows:
1) Did you also apply linear_warm_up and linear_decay strategy?
2) What kind of loss did you adopt? Pair-wise hinge loss or cross-entropy loss or others? For each relevant doc of a query, how many negative docs did you sample?
3) For the two big datasets, the paper assigns the batch_size to 144. Does it mean 144 (query, pos_doc, neg_doc) or 144 (query, doc) {which means 72(query, pos_doc, neg_doc)s}?

Thanks for any possible help.

**Comment by Albert-Ma** [comment]
1. When fine-tuning on MS MARCO document ranking task, yes, we apply the warmup strategy.
2. We use cross-entropy loss with 11 negative docs for one query every epoch, and then use the early-stop technique to choose the best checkpoint.
3. It means 144 (query, doc) which is an instance fed to the model.

**Comment by MooreAndMoore** [comment]
> 1. When fine-tuning on MS MARCO document ranking task, yes, we apply the warmup strategy.
> 2. We use cross-entropy loss with 11 negative docs for one query every epoch, and then use the early-stop technique to choose the best checkpoint.
> 3. It means 144 (query, doc) which is an instance fed to the model.

Thanks for your reply. Could you please explain more about the early-stopping strategy? 
(1) Split the training dataset into a training set and a validation set. (Maybe what's the proportion? Or just use the testing dataset as the so-called validation set)
(2) Train.
(3) Validate. (What's the validation frequency? After each epoch or every 100k steps or others?; Also, what's the validation metric? Use cross-entropy or accuracy or MRR or others?)

**Comment by Albert-Ma** [comment]
> > 1. When fine-tuning on MS MARCO document ranking task, yes, we apply the warmup strategy.
> > 2. We use cross-entropy loss with 11 negative docs for one query every epoch, and then use the early-stop technique to choose the best checkpoint.
> > 3. It means 144 (query, doc) which is an instance fed to the model.
> 
> Thanks for your reply. Could you please explain more about the early-stopping strategy?
> (1) Split the training dataset into a training set and a validation set. (Maybe what's the proportion? Or just use the testing dataset as the so-called validation set)
> (2) Train.
> (3) Validate. (What's the validation frequency? After each epoch or every 100k steps or others?; Also, what's the validation metric? Use cross-entropy or accuracy or MRR or others?)

The early-stopping I mean is that stop training after a specific number of epochs when the validation performance can't improve.
We pick the checkpoint with the best validation performance and then do the evaluation on the test set.
(1) The MS MARCO dataset has its split, i.e. training set and dev set.  For other small datasets, we conduct 5-fold cross-validation and describe the details in section 4.4.3 of [PROP](https://arxiv.org/pdf/2010.10137.pdf).
(2) The training loss refer to [here](https://www.microsoft.com/en-us/research/wp-content/uploads/2016/02/cikm2013_DSSM_fullversion.pdf)
(3) We validate the model after fixed steps, e.g. 50k steps. We use MRR@100 for  MS MARCO document ranking dataset.

**Comment by MooreAndMoore** [comment]
> > > 1. When fine-tuning on MS MARCO document ranking task, yes, we apply the warmup strategy.
> > > 2. We use cross-entropy loss with 11 negative docs for one query every epoch, and then use the early-stop technique to choose the best checkpoint.
> > > 3. It means 144 (query, doc) which is an instance fed to the model.
> > 
> > 
> > Thanks for your reply. Could you please explain more about the early-stopping strategy?
> > (1) Split the training dataset into a training set and a validation set. (Maybe what's the proportion? Or just use the testing dataset as the so-called validation set)
> > (2) Train.
> > (3) Validate. (What's the validation frequency? After each epoch or every 100k steps or others?; Also, what's the validation metric? Use cross-entropy or accuracy or MRR or others?)
> 
> The early-stopping I mean is that stop training after a specific number of epochs when the validation performance can't improve.
> We pick the checkpoint with the best validation performance and then do the evaluation on the test set.
> (1) The MS MARCO dataset has its split, i.e. training set and dev set. For other small datasets, we conduct 5-fold cross-validation and describe the details in section 4.4.3 of [PROP](https://arxiv.org/pdf/2010.10137.pdf).
> (2) The training loss refer to [here](https://www.microsoft.com/en-us/research/wp-content/uploads/2016/02/cikm2013_DSSM_fullversion.pdf)
> (3) We validate the model after fixed steps, e.g. 50k steps. We use MRR@100 for MS MARCO document ranking dataset.



Thank you very much for your kind reply.  As in B-PROP, MSMARCO dev set is used as the testing set.  So how did you split the training set and validation set from the original MSMARCO training set? 
Actually, I find it's kind of time-consuming to take MRR@100 as the validation metric, even if only we split the MSMARCO training set with 95:5.  This means about 18000 queries should be validated, i.e. , 1800000 (query, doc) pairs should be processed. So I'm just wondering whether there is any trick to accelerate the validation.

**Comment by Albert-Ma** [comment]
We didn't split an additional "dev set". We just use the full training set to train the model and directly eval on the real dev set (actually we sampled a small dev set containing 1000 queries).
I think the purpose of splitting another fake dev set is to avoid overfitting on the training set. But I think MS MARCO is large enough and not easy to overfit. If I'm not mistaken, other people who experiment with MSMARCO on their paper also eval on the dev set during training. Another example is the GLUE benchmark in NLP, researchers also directly eval on the true dev set.
Moreover, you can eval on the true dev set and then submit the test result to the MSMARCO leaderboard (in an anonymous way if you think is necessary). [BUT REMEMBER FOLLOWING THE FREQUENCY OF SUBMISSION RULE](https://github.com/microsoft/MSMARCO-Document-Ranking-Submissions/#Frequency-of-Submission).

Hope this will help.

**Comment by MooreAndMoore** [comment]
Thanks for your help!
